File: sample_app/src/main/java/uk/co/alt236/btlescan/ui/common/recyclerview/BaseViewBinder.java
package uk.co.alt236.btlescan.ui.common.recyclerview;

import android.content.Context;

public abstract class BaseViewBinder<T extends RecyclerViewItem> {

    private final Context mContext;

    public BaseViewBinder(final Context context) {
        mContext = context;
    }

    public abstract void bind(BaseViewHolder<T> holder, T item);

    public abstract boolean canBind(RecyclerViewItem item);

    protected Context getContext() {
        return mContext;
    }
}

File: sample_app/src/main/java/uk/co/alt236/btlescan/ui/common/recyclerview/BaseViewBinder.kt
package uk.co.alt236.btlescan.ui.common.recyclerview

import android.content.Context
import androidx.annotation.StringRes
import uk.co.alt236.btlescan.R

abstract class BaseViewBinder<T : RecyclerViewItem>(protected val context: Context) {
    abstract fun bind(holder: BaseViewHolder<T>, item: T)
    abstract fun canBind(item: RecyclerViewItem): Boolean

    protected fun getString(@StringRes id: Int): String {
        return context.getString(id)
    }

    protected fun getString(@StringRes resId: Int, vararg formatArgs: Any?): String {
        return context.getString(resId, *formatArgs)
    }

    protected fun getQuotedString(vararg formatArgs: Any?): String {
        return getString(R.string.formatter_single_quoted_string, *formatArgs)
    }
}